### PR TITLE
fix: Show currently-installed drivers, too

### PR DESCRIPTION
**Pull request by mmstick**
Screenshot from 2022-02-28 19-58-03

### CODE REVIEW

**Comment by jackpot51** [comment]
@mmstick we only want to show the latest driver that supports the current hardware, along with the driver currently installed.

**Review by jackpot51 — APPROVED** [review]

**Comment by XV-02** [comment]
With Kudu6's RTX 3060, this behaves as expected. The currently installed drivers are displayed, and the most recent driver shows. After installing the most recent driver, only that driver displays (as installed). This is true across 20.04 and 21.10

On Thelio, behaviour is _not_ as desired. 
- With 20.04 both driver versions display. After installation of the latest driver, two entries for the driver appear - both of which show the driver as installed.
- With 21.10 and the System76 Nvidia driver package **not** installed, the option to install the nvidia driver doesn't appear in Pop!_Shop at all.
- With 21.10 with the System76 Nvidia driver package **installed**, pop shop shows the installed driver version, but only a single entry.

**Comment by mmstick** [comment]
@XV-02 Does the Thelio have any output from `ubuntu-drivers list`?

**Comment by XV-02** [comment]
@mmstick Yes, passing `ubuntu-drivers list` results in output. 
```
nvidia-driver-470-server, (kernel modules provided by nvidia-dkms-470-server)
nvidia-driver-470, (kernel modules provided by nvidia-dkms-470)
nvidia-driver-510, (kernel modules provided by nvidia-dkms-510)
```

And, just to provide a visual; this is after a clean install of 21.10 with the nvidia drivers (470 from the iso), upgrading Pop!_Shop to the required branch, and then installing the 510 driver. After rebooting, this is what's displayed in Pop!_Shop.

Screenshot from 2022-03-02 17-26-31

**Comment by mmstick** [comment]
Are any 470 packages still installed?

**Comment by XV-02** [comment]
`apt list --installed | grep 470` returns `libxnvctrl0` and `nvidia-settings` as having only the 470 versions present, and `apt clean && apt autoremove` didn't clean them up.

**Review by jackpot51 — DISMISSED** [review]

**Comment by XV-02** [comment]
With the most recent tweak, this now only shows the currently installed driver. With a fresh install, it displayed only 470, even though 510 was available and appropriate. Once 510 was installed manually, it only displayed 510.

**Review by jackpot51 — DISMISSED** [review]

**Comment by XV-02** [comment]
With 470 installed, 510 showed as installable. After installing 510, a second entry for 510 appeared (both entries show as installed), however only a single 510 entry was present after a reboot.

**Comment by XV-02** [comment]
On Thelio with GT-710; 470 shows up as installable, and once installed shows duplicate entries. On reboot, no nvidia driver information is visible in Pop_Shop. Positively, at no point is 510 shown as installable.

**Comment by jackpot51** [comment]
@mmstick is there something you need to fix the issues @XV-02 brought up?

**Comment by mmstick** [comment]
Not sure where this duplicate entry comes from. The appcenter stores every package in a hashmap so duplicate listings shouldn't be possible.

**Comment by jackpot51** [comment]
@mmstick maybe leaking GTK widgets?

**Comment by linuxgnuru** [comment]
I've only seen this if pop shop was currently open when the driver was installed; once pop shop was closed and reopened the duplicate entries disappeared.

**Comment by mmstick** [comment]
@XV-02 I've found a fix for the issue that simply keeps track of what IDs have already been added to the `AppUpdateList`, and preventing the shop from re-adding something it already added before.

**Review by jackpot51 — APPROVED** [review]

**Comment by XV-02** [comment]
So far, looking good on laptops. Next test will be desktop and then 710 for a specific sanity check there.

**Review by n3m0-22 — COMMENTED** [review]
**21.10** this will work with ```nvidia-driver-470``` installed

21 10_470_installed_working_pop_shop

but not with  ```nvidia-driver-510```

20 10_510_installed_not_working

**22.04** I could not get this to work, but that may have to do with the ```nvidia-driver-470``` issue that is currently being worked on here:
https://github.com/pop-os/nvidia-graphics-drivers-470/pull/1

So far I have found no regressions caused by this upgrade.

**Comment by XV-02** [comment]
Looking at 20.04, it seems not to function as intended. One 21.10 and 22.04, both the current installed driver, and the driver upgrade display (That is, with 470 installed, 470 is listed as installed and 510 is shown as available). Once upgraded, the old driver is no longer seen, and a single entry for the current driver is displayed.

On 20.04, with 470 installed, the 510 driver shows as a valid upgrade, but 470 is not shown as installed. Once 510 is installed, however, it correctly displays the 510 driver as installed.

**Comment by XV-02** [comment]
With the GT 710,0 only 470 is shown with this branch present. Without the branch and with 470 installed, it shows 390 as an option for installation and 470 is not displayed.

Unfortunately, if the 510 driver is installed, it does not display in Pop Shop. However the 470 driver is shown as available for install..

TL;DR - with a GT 710, the 470 driver correctly displays as installed or installable, but no other driver is displayed even if installed.

**Comment by leviport** [comment]
>if the 510 driver is installed, it does not display in Pop Shop

I believe this is functioning as designed

**Review by XV-02 — APPROVED** [review]
This is functioning to a sufficient level, and it's idiosyncrasies no longer seem to be legitimate blockers. GT 710 behaviour in particular is good. We're calling it good.
